src/modules/zfshostid/main.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# === This file is part of Calamares - <https://calamares.io> ===
+#
+#   SPDX-FileCopyrightText: 2022 Anke Boersma <demm@kaosx.us>
+#   SPDX-License-Identifier: GPL-3.0-or-later
+#
+#   Calamares is Free Software: see the License-Identifier above.
+#
+
+import os
+import shutil
+
+import libcalamares
+from libcalamares.utils import gettext_path, gettext_languages
+
+
+import gettext
+_ = gettext.translation("calamares-python",
+                        localedir=libcalamares.utils.gettext_path(),
+                        languages=libcalamares.utils.gettext_languages(),
+                        fallback=True).gettext
+
+
+def pretty_name():
+    return _("Copying zfs generated hostid.")
+
+
+def run():
+
+    zfs = libcalamares.globalstorage.value("zfsDatasets")
+    root_mount_point = libcalamares.globalstorage.value("rootMountPoint")
+
+    if zfs:
+        hostid_source = '/etc/hostid'
+        hostid_destination = '{!s}/etc/hostid'.format(root_mount_point)
+
+        if os.path.exists(hostid_source):
```

Comment:
I am not sure this "if" is needed.  It causes the module to fail without giving an error/warning.

If you remove it, the file being missing should cause an exception which will then write a warning per your exception handler.